pwa-wp.php

```diff
@@ -41,4 +41,40 @@ function pwawp_init() {
 
 	$wp_web_app_manifest = new WP_Web_App_Manifest();
 	$wp_web_app_manifest->init();
+
+	// These could be in ABSPATH . WPINC . '/script-loader.php' file.
+	/** WordPress Service Workers Class */
+	require PWAWP_PLUGIN_DIR . '/wp-includes/class.wp-service-workers.php';
+
+	/** WordPress Scripts Functions */
+	require PWAWP_PLUGIN_DIR . '/wp-includes/functions.wp-service-workers.php';
```

Comment:
Changing the filename. However, I was following the naming conventions of `WP_Scripts` instead since the logic is similar. Isn't it be confusing to reuse the methods and logic of `WP_Scrips` and at the same time follow the naming conventions and hierarchy of REST API? Or are we generally trying to follow the logic of the REST API and can expect more similarities to the REST API? Or perhaps the `WP_Scripts` and `WP_Styles` naming conventions are to be changed at some point, too ( e.g. `class-*` instead of `class.*`)? Thoughts?

Reply:
Guessing that following the REST API naming conventions is relevant since with Service Workers (as with REST API) we're sending a HTTP response not just managing the scripts as in case of `WP_Script`.

Reply:
The REST API is newer to core, whereas the Dependencies system is older. So generally the newer stuff has better naming conventions than the older stuff. That's why I looked to the REST API for precedence.

Reply:
Good to know, thanks for clarifying.